klient/conf/main_test.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conf
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"k8s.io/client-go/util/homedir"
+)
+
+func TestMain(m *testing.M) {
+	setup()
+	code := m.Run()
+	teardown()
+	os.Exit(code)
+}
+
+func setup() {
+	home := homedir.HomeDir()
+
+	path := filepath.Join(home, "test", ".kube", "config")
+
+	// check if file exists
+	_, err := os.Stat(path)
+	// create file if not exists
+	if os.IsNotExist(err) {
+		err = os.MkdirAll(home+"/test/.kube", 0777)
```

Comment:
🤔 this is a test so it's probably not that big of a deal, just preventing yet-another-OS-specific implementation that could be easily avoided 😄 

```suggestion
	kubeconfigdir := filepath.Join(home, "test", ".kube")
	kubeconfigpath := filepath.Join(kubeconfigdir, "config")

	// check if file exists
	_, err := os.Stat(kubeconfigpath)
	// create file if not exists
	if os.IsNotExist(err) {
		err = os.MkdirAll(kubeconfigdir, 0777)
```